utils/logger.ts

```diff
@@ -0,0 +1,42 @@
+export abstract class CLELogger {
+  abstract debug(...args: any[]): void
+  abstract info(...args: any[]): void
+  abstract warn(...args: any[]): void
+  abstract error(...args: any[]): void
+  abstract log(...args: any[]): void
+  abstract write(msg: string): void
+}
+
+class Logger implements CLELogger {
+  private writeCache: string[] = []
+  debug(...args: any[]) {
+    console.debug(...args)
+  }
+  info(...args: any[]) {
+    console.info(...args)
+  }
+  warn(...args: any[]) {
+    console.warn(...args)
+  }
+  error(...args: any[]) {
+    console.error(...args)
+  }
+  log(...args: any[]) {
+    console.log(...args)
+  }
+  write(msg: string) {
+    if (msg === '\n') {
+      this.debug(this.writeCache.join(''))
```

Comment:
Not sure if this.debug will add extra contents to the logging. If it added "debug" to the beginning of the log, maybe use this.log?